src/server/string_family.cc

```diff
@@ -325,16 +325,23 @@ OpStatus SetCmd::Set(const SetParams& params, string_view key, string_view value
 
   VLOG(2) << "Set " << key << "(" << db_slice.shard_id() << ") ";
 
-  if (params.how == SET_IF_EXISTS) {
-    auto [it, expire_it] = db_slice.FindExt(params.db_index, key);
-
-    if (IsValid(it)) {  // existing
-      return SetExisting(params, it, expire_it, value);
+  if (params.IsConditionalSet()) {
```

Comment:
curious, why did this require a change? did you change functionality here?

Reply:
I want early exit, in this case we use to test if the key exists, so the case where we have SET_IF_EXISTS we would set it and for the other case we can just skip, so it would ensure that the apply policy is correct, just seems safer to me.